ethereum/rpc/backend/backend.go

```diff
@@ -172,18 +177,18 @@ func (e *EVMBackend) GetTendermintBlockByNumber(blockNum types.BlockNumber) (*tm
 		height = 1
 	default:
 		if blockNum < 0 {
-			err := errors.Errorf("incorrect block height: %d", height)
-			return nil, err
-		} else if height > int64(currentBlockNumber) {
+			return nil, errors.Errorf("cannot fetch a negative block height: %d", height)
+		}
+		if height > int64(currentBlockNumber) {
+			// TODO: should we set height = int64(currentBlockNumber) ?
```

Comment:
I have done a quick check and it seems that infura would return an empty response so current behavior is fine